testsuite/oidc/test_client.py

```diff
@@ -0,0 +1,187 @@
+"""OIDC test client wrapper for easier testing"""
+
+from dataclasses import dataclass, field
+from typing import Optional
+import jwt as jwt_lib
+from keycloak import KeycloakOpenID
+
+from testsuite.httpx.auth import HttpxOidcClientAuth
+from testsuite.oidc import Token
+from testsuite.kuadrant.extensions.oidc_policy import Provider
+
+
+@dataclass
+class ClientConfig:  # pylint: disable=too-many-instance-attributes
+    """Configuration for creating OIDC test clients"""
+
+    client_id: str
+    client_type: str  # "confidential", "public", "service"
```

Comment:
```suggestion
    client_type: Literal["confidential", "public", "service"]
```
if these are the only options